evmcore/tx_pool_test.go

```diff
@@ -391,6 +392,28 @@ func deriveSender(tx *types.Transaction) (common.Address, error) {
 	return types.Sender(types.HomesteadSigner{}, tx)
 }
 
+// setMinTip is a helper function that updates the minimum tip required by the
+// transaction pool for a new transaction, and drops all transactions below
+// this threshold.
+func setMinTip(pool *TxPool, price *big.Int) {
+	pool.mu.Lock()
+	defer pool.mu.Unlock()
+
+	old := pool.minTip
+	pool.minTip = price
+	// if the min miner fee increased, remove transactions below the new threshold
+	if price.Cmp(old) > 0 {
+		// pool.priced is sorted by GasFeeCap, so we have to iterate through pool.all instead
+		drop := pool.all.RemotesBelowTip(price)
+		for _, tx := range drop {
+			pool.removeTx(tx.Hash(), true)
+		}
+		pool.priced.Removed(len(drop))
+	}
+
+	log.Info("Transaction pool price threshold updated", "price", price)
+}
```

Comment:
I am not sure, but I think once we remove the SetGasPrice method from the production code, we no longer need `TestTransactionPoolUnderpricing*` tests (which are the only usage of setMinTip()).

Reply:
I agree, the `repricing` and `underpricing` tests should be refactor and `setGasPrice` removed. But this should be done in a follow-up (https://github.com/0xsoniclabs/sonic-admin/issues/207) as it is not the focus of this PR.